### PR TITLE
fix(tensorrt_yolo): multi gpu on component container

### PR DESCRIPTION
**Pull request by kaancolak**
Signed-off-by: Kaan Colak <kcolak@leodrive.ai>

When working with tensorrt_yolo on multiple GPU device without compostable node container everything works fine. 

But when I use `tensorrt_ yolo` as a component nodelet, if `gpu_device` is set to another then `0` `tensorrt_yolo` gives seg fault. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kaancolak** [comment]
Hello, @wep21 Nithimatsu-san and, @yukkysaito Saito-san,

Could you review this PR? Do you have any opinion or alternative solution?

File: perception/tensorrt_yolo/src/nodelet.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ TensorrtYoloNodelet::TensorrtYoloNodelet(const rclcpp::NodeOptions & options)
   std::string calib_image_directory = declare_parameter("calib_image_directory", "");
   std::string calib_cache_file = declare_parameter("calib_cache_file", "");
   std::string mode = declare_parameter("mode", "FP32");
-  int gpu_device_id = declare_parameter("gpu_id", 0);
+  gpu_device_id_ = declare_parameter("gpu_id", 0);
```
Comment (xmfcx) [inline perception/tensorrt_yolo/src/nodelet.cpp:54]: Can you also update the readme.md with this new parameter?

File: perception/tensorrt_yolo/src/nodelet.cpp [hunk 2]
```diff
@@ -134,6 +134,10 @@ void TensorrtYoloNodelet::callback(const sensor_msgs::msg::Image::ConstSharedPtr
 
   tier4_perception_msgs::msg::DetectedObjectsWithFeature out_objects;
 
+  if (!yolo::set_cuda_device(gpu_device_id_)) {
+    RCLCPP_ERROR(this->get_logger(), "Given GPU not exist or suitable");
```
Comment (wep21) [inline perception/tensorrt_yolo/src/nodelet.cpp:138]: Is it fine not to return here even though gpu device id is invalid?

Comment (kaancolak) [reply perception/tensorrt_yolo/src/nodelet.cpp:138]: Thanks, @wep21. Sorry, it's my fault. Without return, if the given GPU is invalid, tensorrt_yolo will work over GPU 0.

**Review by wep21 — APPROVED** [review]
LGTM
@xmfcx @yukkysaito Do you have another comment?

**Review by xmfcx — APPROVED** [review]

**Pull request merged**